packages/code-du-travail-modeles/src/publicodes/IndemniteLicenciementPublicodes.ts

```diff
@@ -16,10 +25,94 @@ class IndemniteLicenciementPublicodes
     };
     super(
       rules,
-      PublicodesDefaultRules[PublicodesSimulator.INDEMNITE_LICENCIEMENT]
+      PublicodesDefaultRules[PublicodesSimulator.INDEMNITE_LICENCIEMENT],
+      idcc as SupportedCcIndemniteLicenciement
     );
   }
 
+  getMissingArg(
+    args: Record<string, string | undefined>,
+    names: string[]
+  ): string | undefined {
+    let missingArg: string | undefined = undefined;
+    names.some((name) => {
+      if (!args[name]) {
+        missingArg = name;
+        return false;
+      }
+      return true;
+    });
+    return missingArg;
+  }
+
+  throwMissingArg(
+    name: string
+  ): PublicodesData<PublicodesIndemniteLicenciementResult> {
+    return {
+      missingArgs: [
+        {
+          indice: 0,
+          name,
+          rawNode: {
+            nom: name,
+          },
+        },
+      ],
+      result: { value: 0 },
+      situation: [],
+    };
+  }
+
+  calculate(
+    args: Record<string, string | undefined>,
+    targetRule?: string
+  ): PublicodesData<PublicodesIndemniteLicenciementResult> {
+    let newArgs = args;
+    const missingArg = this.getMissingArg(args, [
+      "contrat salarié . indemnité de licenciement . date d'entrée",
+      "contrat salarié . indemnité de licenciement . date de sortie",
+      "contrat salarié . indemnité de licenciement . date de notification",
+    ]);
+    if (missingArg) {
+      return this.throwMissingArg(missingArg);
+    }
+    if (
+      !args["contrat salarié . indemnité de licenciement . ancienneté en année"]
+    ) {
+      const s = new SeniorityFactory().create(this.idcc);
+      const { value, extraInfos }: SeniorityResult = s.computeSeniority(
+        s.mapSituation(args)
+      );
+      newArgs = {
+        ...newArgs,
+        "contrat salarié . indemnité de licenciement . ancienneté conventionnelle en année":
+          value.toString(),
+        "contrat salarié . indemnité de licenciement . ancienneté en année":
+          value.toString(),
+        ...extraInfos,
+      };
+    }
+    if (
+      !args[
+        "contrat salarié . indemnité de licenciement . ancienneté requise en année"
+      ]
+    ) {
+      const s = new SeniorityFactory().create(this.idcc);
+      const { value }: RequiredSeniorityResult = s.computeRequiredSeniority(
+        s.mapRequiredSituation(args)
+      );
+      newArgs = {
+        ...newArgs,
+        "contrat salarié . indemnité de licenciement . ancienneté conventionnelle requise en année":
+          value.toString(),
+        "contrat salarié . indemnité de licenciement . ancienneté requise en année":
+          value.toString(),
+      };
+    }
+    delete newArgs.absencePeriods;
```

Comment:
pas fan du delete, je préfère une destructuration (principe d'immuabilité)

```suggestion
    const {absencePeriods, ...computedArgs} = newArgs;
```

Reply:
je vais tester ce que tu propose.

Reply:
Le soucis c'est que typescript me dit maintenant que absencePeriods n'est pas utilisé. du coup ca fonctionne pas trop à moins que tu es une idée de gérer le typescript derrière

Reply:
Sans compter que d'un point de vue perf, ca nous fait créer une nouvelle variable qui prend de la mémoire (computedArgs) ce qui n'est pas top